xcube/core/dsio.py

```diff
@@ -420,7 +419,7 @@ def read(self, path: str, **kwargs) -> xr.Dataset:
             if 'region_name' in kwargs:
                 client_kwargs['region_name'] = kwargs.pop('region_name')
```

Comment:
What about the lines above? Why do they not apply to write()? Or why are they in read(), not in write()?

Please check and adapt code so processing of args & kwargs in similar in read() and write()